src/statements.md

```diff
@@ -65,6 +66,25 @@ inference. Any variables introduced by a variable declaration are visible
 from the point of declaration until the end of the enclosing block scope,
 except when they are shadowed by another variable declaration.
 
+
+### `let else` statements
+
+> **<sup>Syntax</sup>**\
+> _LetElseStatement_ :\
+> &nbsp;&nbsp; [_OuterAttribute_]<sup>\*</sup> `let` [_PatternNoTopAlt_]
+>     ( `:` [_Type_] )<sup>?</sup> `=` [_Expression_] `else` [_BlockExpression_] `;`
+
+A *`let else` statement* introduces a new set of [variables], given by a
+refutable [pattern]. The pattern is followed optionally by a type
```

Comment:
Is this true for nested type values? I am a little unsure how the type annotation feature works

```rust
// If it were this, this documentation statement would be true:
let Ok(v): Result<u32, _> = expr else { };

// _I thought_ it as supposed to be able to be like this though, but maybe I am wrong:
let Ok(v: u32) = expr else { };
```

Reply:
No, let-else has the same type annotation syntax as let, not sure where you got that idea from.

Reply:
Oh sorry I thought that was part of the upcoming "type ascription" for some reason.